### DISCUSSION TITLE
Picking a line segment in PathLayer or TripsLayer

### DISCUSSION THREAD

**Discussion by jlumsden-mts** [visgl/deck.gl#7772] [Q&A]
Hello, I'd like to try and enable picking particular line segments (rather than an entire path) in the PathLayer by overriding some of the picking functionality. 

If we limit the total number of paths we should be able to use some of the 24-bit color picking range to encode a vertex index. Say 8-bits for up to 255 paths and 16-bits for up to 65535 vertexes per line or some other combination.

I can see that under the hood in the PathLayer there is an instancePickingColor per vertex but when I override encodePickingColor it is only called per path and the vertex arrays are filled from the returned color.

Is it possible to adjust the behavior so I can manipulate the instancePickingColor per vertex and if so how is this done? 

I've also considered doing this on the CPU by projecting the selected path but this is a bit more challenging when the data has elevation (cannot assume sea level) and the user clicks in the middle of a line (away from any vertexes).

**Comment by Pessimistress** [comment]
Assigning picking color per vertex will unlikely work, since the color will be interpreted along the segment. If you don't care much about the corners, you can use the `LineLayer` to render your data.

**Reply by jlumsden-mts** [reply]
Thanks for your response.

I think with careful selection of colours, using a 'zig-zag' approach some of that could be mitigated:
Red channel for path index, green and blue for vertex index
(0,0,0) to (0,0,255), next is (0,1,255) then back down to (0,1,0) and repeat 

I take your point though that the LineLayer already keeps each segment separate so we may be able to use a transparent line layer to pick from either all or the time or as required (two pass picking) so we'll give that a go.

Thanks again!
